src/ve_renogy_rover/rover_service.py

```diff
@@ -204,34 +210,45 @@ def try_(func: Callable[[], Any]):
 
         def solar_power():
             v = try_(rover.solar_voltage)
-            i = try_(rover.solar_current)
+            i = try_(rover.charging_current)
             if v is None or i is None:
                 return None
             return v * i
 
-        updates = {
-            path: value
-            for path, value in {
-                "/Pv/V": try_(rover.solar_voltage),
-                "/Pv/I": try_(rover.solar_current),
-                "/Yield/Power": solar_power(),
-                "/Dc/0/Voltage": try_(rover.battery_voltage),
-                "/Dc/0/Current": try_(rover.charging_current),
-                "/Link/TemperatureSense": try_(rover.battery_temperature),
-                "/Link/TemperatureSenseActive": True,
-                "/History/Daily/0/Yield": try_(rover.power_generation_today),
-                "/History/Daily/0/MaxPower": try_(rover.max_charging_power_today),
-            }.items()
-            if value is not None  # Don't update paths that raise an exception (if any)
-        }
-
-        if (operation_mode := OperationMode.from_rover(try_(rover.charging_state))) is not None:
-            updates["/MppOperationMode"] = operation_mode.value
-
-        if (state := State.from_rover(try_(rover.charging_state))) is not None:
-            updates["/State"] = state.value
-
-        with self._dbusservice as s:
+        def charging_current():
+            p = try_(rover.charging_power)
+            v = try_(rover.battery_voltage)
+            if p and v and v > 0:
+                return p / v
+            return 0.0
+
+        try:
+            updates = {
+                path: value
+                for path, value in {
+                    "/Pv/V": try_(rover.solar_voltage),
+                    "/Pv/I": try_(rover.charging_current),
+                    "/Yield/Power": solar_power(),
+                    "/Dc/0/Voltage": try_(rover.battery_voltage),
+                    "/Dc/0/Current": charging_current(),
+                    "/Link/TemperatureSense": try_(rover.battery_temperature),
+                    "/Link/TemperatureSenseActive": True,
+                    "/History/Daily/0/Yield": try_(rover.power_generation_today) or 0.0,
+                    "/History/Daily/0/MaxPower": (try_(rover.max_charging_power_today) or 0.0) / 1000.0,
+                }.items()
+                if value is not None  # Don't update paths that raise an exception (if any)
+            }
+
+            if (operation_mode := OperationMode.from_rover(try_(rover.charging_state))) is not None:
+                updates["/MppOperationMode"] = operation_mode.value
+
+            if (state := State.from_rover(try_(rover.charging_state))) is not None:
+                updates["/State"] = state.value
+        except Exception as e:
+            logging.exception(f"Error updating path values: {e}")
+            return True
```

Comment:
We can do this differently now since this shouldn't raise anymore. Maybe issue a warning for paths that return a `None` value for the dict comprehension above.